hybrid/compartment_examples/comp_setup/main.c

```diff
@@ -113,14 +113,23 @@ add_comp(uint8_t* _start_addr, void (* _comp_fn)())
     ++id;
 }
 
+/*******************************************************************************
+ * Helper Functions
+ ******************************************************************************/
+
 void
 validate_offsets()
 {
-    assert(COMP_SIZE == sizeof(struct comp));
-    assert(COMP_OFFSET_STK_ADDR == offsetof(struct comp, stack_addr));
-    assert(COMP_OFFSET_STK_LEN == offsetof(struct comp, stack_len));
-    assert(COMP_OFFSET_DDC == offsetof(struct comp, ddc));
-    assert(COMP_OFFSET_PCC == offsetof(struct comp, comp_fn));
+    static_assert(COMP_SIZE == sizeof(struct comp),
```

Comment:
I wonder: do `static_assert`s need to be in a function? If not, we can place these at the top-level of the file when it'll be clearer that they're static thingies.

Reply:
Updated in https://github.com/capablevms/cheri-examples/pull/52/commits/b93d6d729375eb268f85b93ba49f7d26bb88dbc3.